aws/resource_aws_fsx_file_system_test.go

```diff
@@ -0,0 +1,365 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/fsx"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSFsxFileSystem_lustreBasic(t *testing.T) {
+	var v fsx.FileSystem
+	resourceName := "aws_fsx_file_system.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: resourceName,
+		Providers:     testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsFsxFileSystemLustreBasic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckFileSystemExists(resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "type", "LUSTRE"),
+					resource.TestCheckResourceAttr(resourceName, "capacity", "3600"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"kms_key_id"},
+			},
+		},
+	})
+}
+
+func TestAccAWSFsxFileSystem_lustreConfig(t *testing.T) {
+	var v fsx.FileSystem
+	resourceName := "aws_fsx_file_system.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: resourceName,
+		Providers:     testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsFsxFileSystemLustreConfigOpts,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckFileSystemExists(resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "lustre_configuration.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "lustre_configuration.0.import_path", "s3://nasanex"),
+					resource.TestCheckResourceAttr(resourceName, "lustre_configuration.0.chunk_size", "2048"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"kms_key_id"},
```

Comment:
We should not need `ImportStateVerifyIgnore` for this attribute and this should be removed. If this testing does not work otherwise, please let us know if you are having trouble determining the cause.